modules/decrypt-node/src/parse_header_stream.ts

```diff
@@ -36,14 +44,34 @@ export class ParseHeaderStream extends PortableTransformWithType {
     this._headerState = {
       buffer: Buffer.alloc(0),
       headerParsed: false,
+      maxHeaderSize: maxHeaderSize,
     }
   }
 
   _transform(chunk: any, encoding: string, callback: Function) {
-    const { buffer } = this._headerState
+    const { buffer, maxHeaderSize } = this._headerState
     const headerBuffer = Buffer.concat([buffer, chunk])
-    const headerInfo = deserialize.deserializeMessageHeader(headerBuffer)
+
+    /* maxHeaderSize is the maximum
+     * size we MUST attempt
+     * to deserialize.
+     */
+    const headerInfo = deserialize.deserializeMessageHeader(
+      maxHeaderSize && headerBuffer.byteLength > maxHeaderSize
```

Comment:
Let's consolidate this and line 74 into a single resolution of this logic check that we reference both places. That should make this simpler to read and also less likely to diverge in the future.